src/stat.h

```diff
@@ -0,0 +1,37 @@
+#ifndef _STAT_H_
+#define _STAT_H_
+
+#include <stdint.h>
+#include <ev.h>
+
+typedef struct {
+  struct ev_loop *loop;
+  int stats_interval;
+  size_t requests_size;
+  size_t responses_size;
+  uint64_t requests;
+  uint64_t responses;
+  uint64_t query_times_sum;
+  uint64_t connections_opened;
+  uint64_t connections_closed;
+  uint64_t connections_reused;
+  ev_timer stats_timer;
+} stat_t;
+
+void stat_init(stat_t *s, struct ev_loop *loop, int stats_interval);
+
+void stat_request_begin(stat_t *s, size_t req_len);
+
+void stat_request_end(stat_t *s, size_t resp_len, ev_tstamp latency);
+
+void stat_connection_opened(stat_t *s);
+
+void stat_connection_closed(stat_t *s);
+
+void stat_connection_reused(stat_t *s);
+
+void stat_stop(stat_t *s);
```

Comment:
Some nits really:
* stat_init, stat_stop, stat_cleanup are all related to managing the timer event loop and allocation. Can we group those together.
* Can we comment these?

Reply:
Did I understand correctly, that:

- function definitions and declarations should be next to each other in the source files?
Because as far I saw in other cases (dns_server.* and https_client.*) the order is: *_init() other functions and at last *_stop() and *_cleanup()

- I could comment them like the following:
```
stat_init() // initialize statistic struct and start timer if needed
stat_stop() // stop stat timer for gracefull shutdown
stat_cleanup() // print last statistic measurement before program exit
```

Reply:
Yeah, even a file-level comment would be useful. I guess I was first expecting stats to just be something like `if ((cnt++ % 1000) == 0) { SLOG(...); }` so it took a few minutes to appreciate the timer bit. 
I think I'm perhaps being a bit more pedantic here because it was foreign looking code to me to start with.

Feel free to edit this if you think I'm mis-representing anything:
```
// Statistics tracking
//
// Connection, request and latency statistics are accumulated in a struct and output 
// periodically via a libev timer callback.
//
// stat_init() allocates this struct and starts the timer if required.
// stat_stop() stops the timer for shutdown.
// stat_cleanup() prints the final measurement.
// stat_request_(begin|end), stat_connection_(open|closed|reused) update the tallies.
//
```